### PR TITLE
test: Add ExceptionActivity

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**
trace-test-application will have a test screen, where we can generate sample crashes.

APM-3120

### CODE REVIEW

File: trace-test-application/src/main/res/layout/activity_exception.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:padding="20dp"
+    tools:context=".IndexActivity">
```
Comment (daniellevass-bitrise) [inline trace-test-application/src/main/res/layout/activity_exception.xml:8]: nit: this should be .ExceptionActivity?

File: trace-test-application/src/main/res/layout/activity_exception.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        android:id="@+id/btn_exception_main_thread"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Throw exception on main thread" />
+
+    <Button
+        android:id="@+id/btn_exception_thread"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Throw  exception on thread" />
```
Comment (daniellevass-bitrise) [inline trace-test-application/src/main/res/layout/activity_exception.xml:44]: nit you have two space characters between "throw" and "exception".

File: trace-test-application/src/main/java/io/bitrise/trace/testapp/exception/ExceptionActivity.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   * Throws an exception in the main thread. Should be called from the main thread.
+   */
+  private void throwExceptionInMainThread() {
+    throwException("Main thread is not happy!");
+  }
+
+  /**
+   * Creates a new thread and throws an exception.
+   */
+  private void throwExceptionInThread() {
```
Comment (daniellevass-bitrise) [inline trace-test-application/src/main/java/io/bitrise/trace/testapp/exception/ExceptionActivity.java:101]: nit: I'd rename this as `throwExceptionInNewThread` and replace the button text too - it's a bit confusing as you have Main Thread, and then Thread (of which Main Thread is a Thread) - explicitly adding this is a "new" thread would help readability.

File: trace-test-application/src/main/java/io/bitrise/trace/testapp/exception/ExceptionActivity.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    new Thread() {
+      @Override
+      public void run() {
+        super.run();
+        throwException(String.format("Thread with name %s is not happy",
```
Comment (daniellevass-bitrise) [inline trace-test-application/src/main/java/io/bitrise/trace/testapp/exception/ExceptionActivity.java:106]: nit: all your other exception messages end with a `!` character - this should either end with a full stop or other ending character.

**Review by daniellevass-bitrise — APPROVED** [review]
Left a couple of nits to look at. Thank you for your work on this 🎉

File: trace-test-application/src/main/res/layout/activity_exception.xml (hunk 1 above)
Comment (richard-bogdan-bitrise) [reply trace-test-application/src/main/res/layout/activity_exception.xml:8]: It is not needed, removed.

**Pull request merged**